src/semantics/mod.rs

```diff
@@ -58,12 +60,14 @@ pub fn check_module(raw_module: &raw::Module) -> Result<Module, TypeError> {
 }
 
 /// Reduce a term to its normal form
-pub fn normalize(context: &Context, term: &RcTerm) -> Result<RcValue, InternalError> {
-    use syntax::context::Definition;
-
+pub fn normalize(
+    prim_env: &PrimEnv,
+    context: &Context,
```

Comment:
Maybe we could merge these into some combined TcEnv at some stage! Would make passing this stuff around a little bit easier...